`test_ibootbar_set_outlet` can be flaky

The ibootbar integration test `test_ibootbar_set_outlet` can be flaky, succeeding sometimes, but failing others.

Here's an example of it failing in a recent CI run:
```
Run COVERAGE_FILE=.coverage.int python3 -m pytest --cov -m 'integtest'
============================= test session starts ==============================
platform linux -- Python 3.8.16, pytest-7.3.1, pluggy-1.0.0
rootdir: /home/runner/work/socs/socs/tests
configfile: pytest.ini
plugins: twisted-1.14.0, cov-4.0.0, order-1.1.0, dependency-0.5.1, docker-compose-3.2.1
collected 172 items / 67 deselected / 105 selected

common/test_moxa_serial.py ....                                          [  3%]
common/test_pmx.py Caught connection reset on Agent clean up
.Caught connection reset on Agent clean up
.............                                        [ 17%]
integration/test_cryomech_cpa_agent_integration.py ......                [ 22%]
integration/test_hwp_rotation_agent_integration.py .................     [ 39%]
integration/test_ibootbar_agent_integration.py .F..                      [ 42%]
integration/test_ls240_agent_integration.py ....                         [ 46%]
integration/test_ls372_agent_integration.py .........................    [ 70%]
integration/test_ls425_agent_integration.py .......                      [ 77%]
integration/test_pfeiffer_tc400_agent_integration.py .........           [ 85%]
integration/test_pfeiffer_tpg366_agent_integration.py .                  [ 86%]
integration/test_pysmurf_monitor_integration.py ...                      [ 89%]
integration/test_scpi_psu_agent_integration.py .....                     [ 94%]
integration/test_synacc_integration.py .....                             [ 99%]
integration/test_ups_agent_integration.py .                              [100%]

=================================== FAILURES ===================================
___________________________ test_ibootbar_set_outlet ___________________________

wait_for_crossbar = None, start_responder = None, run_agent = None
client = OCSClient('ibootbar')

    @pytest.mark.integtest
    @inlineCallbacks
    def test_ibootbar_set_outlet(wait_for_crossbar, start_responder, run_agent, client):
        outlet_number = 3
        resp = client.set_outlet(outlet=outlet_number, state="on")
        check_resp_success(resp)
    
        # Simulate internal state transition of hardware
        snmp = SNMPTwister(address, port)
        outlet = [("IBOOTPDU-MIB", "outletStatus", outlet_number - 1)]
        yield snmp.set(oid_list=outlet, version=2, setvalue=1, community_name="public")
    
        resp = client.acq.start(test_mode=True)
        resp = client.acq.wait()
    
>       assert resp.session["data"][f"outletStatus_{outlet_number - 1}"]["status"] == 1
E       KeyError: 'outletStatus_2'

integration/test_ibootbar_agent_integration.py:82: KeyError
---------------------------- Captured stdout setup -----------------------------
/home/runner/work/socs/socs/tests
---------------------------- Captured stderr setup -----------------------------
Using "NullReporter" activity reporting method with params  
Scanning "/home/runner/.snmpsim/variation" directory for variation modules... 
Directory "/home/runner/.snmpsim/variation" does not exist 
Scanning "/opt/hostedtoolcache/Python/3.8.16/x64/snmpsim/variation" directory for variation modules... 
A total of 9 modules found in /opt/hostedtoolcache/Python/3.8.16/x64/snmpsim/variation 
Scanning "/opt/hostedtoolcache/Python/3.8.16/x64/share/snmpsim/variation" directory for variation modules... 
Directory "/opt/hostedtoolcache/Python/3.8.16/x64/share/snmpsim/variation" does not exist 
Scanning "/opt/hostedtoolcache/Python/3.8.16/x64/lib/python3.8/site-packages/snmpsim/variation" directory for variation modules... 
Directory "/opt/hostedtoolcache/Python/3.8.16/x64/lib/python3.8/site-packages/snmpsim/variation" does not exist 
Initializing variation modules... 
ERROR Variation module "sql" from "/opt/hostedtoolcache/Python/3.8.16/x64/snmpsim/variation/sql.py" load FAILED: database type not specified 
Variation module "writecache" from "/opt/hostedtoolcache/Python/3.8.16/x64/snmpsim/variation/writecache.py" loaded OK 
Variation module "subprocess" from "/opt/hostedtoolcache/Python/3.8.16/x64/snmpsim/variation/subprocess.py" loaded OK 
Variation module "notification" from "/opt/hostedtoolcache/Python/3.8.16/x64/snmpsim/variation/notification.py" loaded OK 
Variation module "delay" from "/opt/hostedtoolcache/Python/3.8.16/x64/snmpsim/variation/delay.py" loaded OK 
Variation module "error" from "/opt/hostedtoolcache/Python/3.8.16/x64/snmpsim/variation/error.py" loaded OK 
Variation module "numeric" from "/opt/hostedtoolcache/Python/3.8.16/x64/snmpsim/variation/numeric.py" loaded OK 
Variation module "multiplex" from "/opt/hostedtoolcache/Python/3.8.16/x64/snmpsim/variation/multiplex.py" loaded OK 
ERROR Variation module "redis" from "/opt/hostedtoolcache/Python/3.8.16/x64/snmpsim/variation/redis.py" load FAILED: Redis connect parameters not specified 
--- SNMP Engine configuration 
SNMPv3 EngineID: 0x80004fb80566762d617a3138302d36323513c2f700 
  --- Simulation data recordings configuration 
  SNMPv3 Context Engine ID: 0x80004fb80566762d617a3138302d36323513c2f700 
  Scanning "./integration/ibootbar_snmp_data" directory for  *.dump, *.MVC, *.sapwalk, *.snmpwalk, *.snmprec, *.snmprec.bz2 data files... 
    Configuring ./integration/ibootbar_snmp_data/public.snmprec controller 
    SNMPv1/2c community name: public 
    SNMPv3 Context Name: 4c9184f37cff01bcdc32dc486ec36961 or public 
    Configuring ./integration/ibootbar_snmp_data/private.snmprec controller 
    SNMPv1/2c community name: private 
    SNMPv3 Context Name: 2c17c6393771ee3048ae34d6b380c5ec or private 
  --- SNMPv3 USM configuration 
  SNMPv3 USM SecurityName: simulator 
  SNMPv3 USM authentication key: auctoritas, authentication protocol: MD5 
  SNMPv3 USM encryption (privacy) key: privatus, encryption protocol: DES 
  Maximum number of variable bindings in SNMP response: 64 
  --- Transport configuration 
  Listening at UDP/IPv4 endpoint 127.0.0.1:1024, transport ID 1.3.6.1.6.1.1.0 
----------------------------- Captured stdout call -----------------------------
OCSReply: OK : Operation "set_outlet" is currently not running (SUCCEEDED).
  set_outlet[session=0]; status=done without error 0.003479 s ago, took 7.5 s
  messages (3 of 3):
    1682637258.276 Status is now "starting".
    1682637265.787 Set outlet 2 to on
    1682637265.788 Status is now "done".
  other keys in .session: op_code, data
{'session_id': 0, 'op_name': 'set_outlet', 'op_code': 5, 'status': 'done', 'success': True, 'start_time': 1682637258.2764864, 'end_time': 1682637265.7875574, 'data': {}, 'messages': [[1682637258.2764864, 'Status is now "starting".'], [1682637265.7870305, 'Set outlet 2 to on'], [1682637265.7875574, 'Status is now "done".']]}
----------------------------- Captured stderr call -----------------------------
Shutting down variation modules: 
Variation module "sql" shutdown OK 
Variation module "writecache" shutdown OK 
Variation module "subprocess" shutdown OK 
Variation module "notification" shutdown OK 
Variation module "delay" shutdown OK 
Variation module "error" shutdown OK 
Variation module "numeric" shutdown OK 
Variation module "multiplex" shutdown OK 
Variation module "redis" shutdown OK 
Process terminated 
Process Process-2:
Traceback (most recent call last):
  File "/opt/hostedtoolcache/Python/3.8.16/x64/lib/python3.8/site-packages/pysnmp/carrier/asyncore/dispatch.py", line 45, in runDispatcher
    loop(timeout or self.getTimerResolution(),
  File "/opt/hostedtoolcache/Python/3.8.16/x64/lib/python3.8/asyncore.py", line 207, in loop
    poll_fun(timeout, map)
  File "/opt/hostedtoolcache/Python/3.8.16/x64/lib/python3.8/asyncore.py", line 188, in poll2
    readwrite(obj, flags)
  File "/opt/hostedtoolcache/Python/3.8.16/x64/lib/python3.8/asyncore.py", line 123, in readwrite
    obj.handle_error()
  File "/opt/hostedtoolcache/Python/3.8.16/x64/lib/python3.8/asyncore.py", line 108, in readwrite
    obj.handle_read_event()
  File "/opt/hostedtoolcache/Python/3.8.16/x64/lib/python3.8/asyncore.py", line 420, in handle_read_event
    self.handle_read()
  File "/opt/hostedtoolcache/Python/3.8.16/x64/lib/python3.8/site-packages/pysnmp/carrier/asyncore/dgram/base.py", line 170, in handle_read
    self._cbFun(self, transportAddress, incomingMessage)
  File "/opt/hostedtoolcache/Python/3.8.16/x64/lib/python3.8/site-packages/pysnmp/carrier/base.py", line 84, in _cbFun
    self.__recvCallables[recvId](
  File "/opt/hostedtoolcache/Python/3.8.16/x64/lib/python3.8/site-packages/pysnmp/entity/engine.py", line 151, in __receiveMessageCbFun
    self.msgAndPduDsp.receiveMessage(
  File "/opt/hostedtoolcache/Python/3.8.16/x64/lib/python3.8/site-packages/pysnmp/proto/rfc3412.py", line 291, in receiveMessage
    msgVersion = verdec.decodeMessageVersion(wholeMsg)
  File "/opt/hostedtoolcache/Python/3.8.16/x64/lib/python3.8/site-packages/pysnmp/proto/api/verdec.py", line 15, in decodeMessageVersion
    seq, wholeMsg = decoder.decode(
  File "/opt/hostedtoolcache/Python/3.8.16/x64/lib/python3.8/site-packages/pyasn1/codec/ber/decoder.py", line 2003, in __call__
    for asn1Object in streamingDecoder:
  File "/opt/hostedtoolcache/Python/3.8.16/x64/lib/python3.8/site-packages/pyasn1/codec/ber/decoder.py", line 1918, in __iter__
    for asn1Object in self._singleItemDecoder(
  File "/opt/hostedtoolcache/Python/3.8.16/x64/lib/python3.8/site-packages/pyasn1/codec/ber/decoder.py", line 1778, in __call__
    for value in concreteDecoder.valueDecoder(
  File "/opt/hostedtoolcache/Python/3.8.16/x64/lib/python3.8/site-packages/pyasn1/codec/ber/decoder.py", line 654, in valueDecoder
    for chunk in substrateFun(asn1Object, substrate, length, options):
TypeError: <lambda>() takes 3 positional arguments but 4 were given

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/opt/hostedtoolcache/Python/3.8.16/x64/lib/python3.8/multiprocessing/process.py", line 315, in _bootstrap
    self.run()
  File "/opt/hostedtoolcache/Python/3.8.16/x64/lib/python3.8/multiprocessing/process.py", line 108, in run
    self._target(*self._args, **self._kwargs)
  File "/opt/hostedtoolcache/Python/3.8.16/x64/lib/python3.8/site-packages/snmpsim/commands/responder.py", line 916, in main
    transport_dispatcher.runDispatcher()
  File "/opt/hostedtoolcache/Python/3.8.16/x64/lib/python3.8/site-packages/pysnmp/carrier/asyncore/dispatch.py", line 50, in runDispatcher
    raise PySnmpError('poll error: %s' % ';'.join(format_exception(*exc_info())))
pysnmp.error.PySnmpError: poll error: Traceback (most recent call last):
;  File "/opt/hostedtoolcache/Python/3.8.16/x64/lib/python3.8/site-packages/pysnmp/carrier/asyncore/dispatch.py", line 45, in runDispatcher
    loop(timeout or self.getTimerResolution(),
;  File "/opt/hostedtoolcache/Python/3.8.16/x64/lib/python3.8/asyncore.py", line 207, in loop
    poll_fun(timeout, map)
;  File "/opt/hostedtoolcache/Python/3.8.16/x64/lib/python3.8/asyncore.py", line 188, in poll2
    readwrite(obj, flags)
;  File "/opt/hostedtoolcache/Python/3.8.16/x64/lib/python3.8/asyncore.py", line 123, in readwrite
    obj.handle_error()
;  File "/opt/hostedtoolcache/Python/3.8.16/x64/lib/python3.8/asyncore.py", line 108, in readwrite
    obj.handle_read_event()
;  File "/opt/hostedtoolcache/Python/3.8.16/x64/lib/python3.8/asyncore.py", line 420, in handle_read_event
    self.handle_read()
;  File "/opt/hostedtoolcache/Python/3.8.16/x64/lib/python3.8/site-packages/pysnmp/carrier/asyncore/dgram/base.py", line 170, in handle_read
    self._cbFun(self, transportAddress, incomingMessage)
;  File "/opt/hostedtoolcache/Python/3.8.16/x64/lib/python3.8/site-packages/pysnmp/carrier/base.py", line 84, in _cbFun
    self.__recvCallables[recvId](
;  File "/opt/hostedtoolcache/Python/3.8.16/x64/lib/python3.8/site-packages/pysnmp/entity/engine.py", line 151, in __receiveMessageCbFun
    self.msgAndPduDsp.receiveMessage(
;  File "/opt/hostedtoolcache/Python/3.8.16/x64/lib/python3.8/site-packages/pysnmp/proto/rfc3412.py", line 291, in receiveMessage
    msgVersion = verdec.decodeMessageVersion(wholeMsg)
;  File "/opt/hostedtoolcache/Python/3.8.16/x64/lib/python3.8/site-packages/pysnmp/proto/api/verdec.py", line 15, in decodeMessageVersion
    seq, wholeMsg = decoder.decode(
;  File "/opt/hostedtoolcache/Python/3.8.16/x64/lib/python3.8/site-packages/pyasn1/codec/ber/decoder.py", line 2003, in __call__
    for asn1Object in streamingDecoder:
;  File "/opt/hostedtoolcache/Python/3.8.16/x64/lib/python3.8/site-packages/pyasn1/codec/ber/decoder.py", line 1918, in __iter__
    for asn1Object in self._singleItemDecoder(
;  File "/opt/hostedtoolcache/Python/3.8.16/x64/lib/python3.8/site-packages/pyasn1/codec/ber/decoder.py", line 1778, in __call__
    for value in concreteDecoder.valueDecoder(
;  File "/opt/hostedtoolcache/Python/3.8.16/x64/lib/python3.8/site-packages/pyasn1/codec/ber/decoder.py", line 654, in valueDecoder
    for chunk in substrateFun(asn1Object, substrate, length, options):
;TypeError: <lambda>() takes 3 positional arguments but 4 were given
caused by <class 'TypeError'>: <lambda>() takes 3 positional arguments but 4 were given
=============================== warnings summary ===============================
../../../../../../opt/hostedtoolcache/Python/3.8.16/x64/lib/python3.8/site-packages/so3g/spt3g_internal/core/load_pybindings.py:21
  /opt/hostedtoolcache/Python/3.8.16/x64/lib/python3.8/site-packages/so3g/spt3g_internal/core/load_pybindings.py:21: DeprecationWarning: the imp module is deprecated in favour of importlib; see the module's documentation for alternative uses
    import imp, os, sys

../../../../../../opt/hostedtoolcache/Python/3.8.16/x64/lib/python3.8/site-packages/sotodlib/io/g3tsmurf_db.py:8
  /opt/hostedtoolcache/Python/3.8.16/x64/lib/python3.8/site-packages/sotodlib/io/g3tsmurf_db.py:8: MovedIn20Warning: The ``declarative_base()`` function is now available as sqlalchemy.orm.declarative_base(). (deprecated since: 2.0) (Background on SQLAlchemy 2.0 at: https://sqlalche.me/e/b8d9)
    Base = declarative_base()

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html

---------- coverage: platform linux, python 3.8.16-final-0 -----------
Name                                                                              Stmts   Miss  Cover
-----------------------------------------------------------------------------------------------------
/home/runner/work/socs/socs/socs/Lakeshore/Lakeshore240.py                          220     72    67%
/home/runner/work/socs/socs/socs/Lakeshore/Lakeshore372.py                          688    313    55%
/home/runner/work/socs/socs/socs/Lakeshore/Lakeshore425.py                           30      1    97%
/home/runner/work/socs/socs/socs/__init__.py                                          3      0   100%
/home/runner/work/socs/socs/socs/agents/acu/agent.py                                644    569    12%
/home/runner/work/socs/socs/socs/agents/acu/drivers.py                              234    222     5%
/home/runner/work/socs/socs/socs/agents/bluefors/agent.py                           202    163    19%
/home/runner/work/socs/socs/socs/agents/cryomech_cpa/agent.py                       179     16    91%
/home/runner/work/socs/socs/socs/agents/fts_aerotech/agent.py                       188    157    16%
/home/runner/work/socs/socs/socs/agents/holo_fpga/agent.py                           87     87     0%
/home/runner/work/socs/socs/socs/agents/holo_synth/agent.py                          95     95     0%
/home/runner/work/socs/socs/socs/agents/hwp_encoder/agent.py                        258    222    14%
/home/runner/work/socs/socs/socs/agents/hwp_picoscope/agent.py                      120    120     0%
/home/runner/work/socs/socs/socs/agents/hwp_picoscope/drivers/class_ps3000a.py      195    195     0%
/home/runner/work/socs/socs/socs/agents/hwp_rotation/agent.py                       238     48    80%
/home/runner/work/socs/socs/socs/agents/hwp_rotation/drivers/pid_controller.py      189     14    93%
/home/runner/work/socs/socs/socs/agents/ibootbar/agent.py                           167     43    74%
/home/runner/work/socs/socs/socs/agents/ifm_sbn246_flowmeter/agent.py                74     74     0%
/home/runner/work/socs/socs/socs/agents/labjack/agent.py                            269    269     0%
/home/runner/work/socs/socs/socs/agents/lakeshore240/agent.py                       148     31    79%
/home/runner/work/socs/socs/socs/agents/lakeshore336/agent.py                       410    348    15%
/home/runner/work/socs/socs/socs/agents/lakeshore370/agent.py                       381    316    17%
/home/runner/work/socs/socs/socs/agents/lakeshore372/agent.py                       689    226    67%
/home/runner/work/socs/socs/socs/agents/lakeshore425/agent.py                       136     18    87%
/home/runner/work/socs/socs/socs/agents/magpie/agent.py                             470    401    15%
/home/runner/work/socs/socs/socs/agents/meinberg_m1000/agent.py                     130    104    20%
/home/runner/work/socs/socs/socs/agents/ocs_plugin_so.py                              5      0   100%
/home/runner/work/socs/socs/socs/agents/pfeiffer_tc400/agent.py                     123     17    86%
/home/runner/work/socs/socs/socs/agents/pfeiffer_tc400/drivers.py                    55      8    85%
/home/runner/work/socs/socs/socs/agents/pfeiffer_tpg366/agent.py                     97     17    82%
/home/runner/work/socs/socs/socs/agents/pysmurf_controller/agent.py                 322    249    23%
/home/runner/work/socs/socs/socs/agents/pysmurf_monitor/agent.py                    111     11    90%
/home/runner/work/socs/socs/socs/agents/scpi_psu/agent.py                            98      7    93%
/home/runner/work/socs/socs/socs/agents/scpi_psu/drivers.py                          47     10    79%
/home/runner/work/socs/socs/socs/agents/smurf_crate_monitor/agent.py                154    135    12%
/home/runner/work/socs/socs/socs/agents/smurf_file_emulator/agent.py                393    330    16%
/home/runner/work/socs/socs/socs/agents/smurf_stream_simulator/agent.py             162    126    22%
/home/runner/work/socs/socs/socs/agents/smurf_timing_card/agent.py                   56     56     0%
/home/runner/work/socs/socs/socs/agents/suprsync/agent.py                            93     93     0%
/home/runner/work/socs/socs/socs/agents/synacc/agent.py                             122      9    93%
/home/runner/work/socs/socs/socs/agents/tektronix3021c/agent.py                      80     61    24%
/home/runner/work/socs/socs/socs/agents/tektronix3021c/drivers.py                     8      3    62%
/home/runner/work/socs/socs/socs/agents/thorlabs_mc2000b/agent.py                   149    149     0%
/home/runner/work/socs/socs/socs/agents/ups/agent.py                                126     42    67%
/home/runner/work/socs/socs/socs/agents/vantagepro2/agent.py                        107     90    16%
/home/runner/work/socs/socs/socs/agents/vantagepro2/drivers.py                      162    144    11%
/home/runner/work/socs/socs/socs/agents/wiregrid_actuator/agent.py                  437    437     0%
/home/runner/work/socs/socs/socs/agents/wiregrid_actuator/drivers/Actuator.py       245    245     0%
/home/runner/work/socs/socs/socs/agents/wiregrid_actuator/drivers/DigitalIO.py       80     80     0%
/home/runner/work/socs/socs/socs/agents/wiregrid_actuator/limitswitch_config.py       1      1     0%
/home/runner/work/socs/socs/socs/agents/wiregrid_actuator/stopper_config.py           1      1     0%
/home/runner/work/socs/socs/socs/agents/wiregrid_encoder/agent.py                   154    134    13%
/home/runner/work/socs/socs/socs/agents/wiregrid_encoder/drivers.py                 156    132    15%
/home/runner/work/socs/socs/socs/agents/wiregrid_kikusui/agent.py                   333    300    10%
/home/runner/work/socs/socs/socs/agents/wiregrid_kikusui/drivers/common.py           23     18    22%
/home/runner/work/socs/socs/socs/agents/xy_stage/agent.py                           146    119    18%
/home/runner/work/socs/socs/socs/common/moxa_serial.py                               97     30    69%
/home/runner/work/socs/socs/socs/common/pmx.py                                      317    103    68%
/home/runner/work/socs/socs/socs/common/prologix_interface.py                        30      2    93%
/home/runner/work/socs/socs/socs/db/suprsync.py                                     185    130    30%
/home/runner/work/socs/socs/socs/plugin.py                                            2      0   100%
/home/runner/work/socs/socs/socs/snmp.py                                             45     13    71%
/home/runner/work/socs/socs/socs/testing/device_emulator.py                         132      8    94%
/home/runner/work/socs/socs/socs/util.py                                              6      0   100%
-----------------------------------------------------------------------------------------------------
TOTAL                                                                             11304   7434    34%

47 empty files skipped.

=========================== short test summary info ============================
FAILED integration/test_ibootbar_agent_integration.py::test_ibootbar_set_outlet - KeyError: 'outletStatus_2'
===== 1 failed, 104 passed, 67 deselected, 2 warnings in 563.64s (0:09:23) =====
Error: Process completed with exit code 1.
```

--- a/requirements.txt
+++ b/requirements.txt
@@ -6,8 +6,9 @@ twisted
 
 # hardware communication
 pyserial
-pysnmp
+pysnmp-lextudio
 pysmi
+pyasn1==0.4.8
 
 # common dependencies - used by multiple agents
 numpy

--- a/setup.py
+++ b/setup.py
@@ -97,11 +97,12 @@ setup(
         'numpy',
         'ocs',
         'pyserial',
-        'pysnmp',
+        'pysnmp-lextudio',
         'pysmi',
         'pyyaml',
         'sqlalchemy>=1.4',
         'twisted',
+        'pyasn1==0.4.8',
     ],
     extras_require={
         'all': all_deps,

--- a/tests/integration/test_ibootbar_agent_integration.py
+++ b/tests/integration/test_ibootbar_agent_integration.py
@@ -11,9 +11,9 @@ from ocs.base import OpCode
 from ocs.testing import create_agent_runner_fixture, create_client_fixture
 from snmpsim.commands import responder
 
-# from twisted.internet.defer import inlineCallbacks  # disabled for issue #434
+from twisted.internet.defer import inlineCallbacks
 
-# from socs.snmp import SNMPTwister  # disabled for issue #434
+from socs.snmp import SNMPTwister
 
 pytest_plugins = "docker_compose"
 
@@ -65,24 +65,22 @@ def test_ibootbar_acq(wait_for_crossbar, start_responder, run_agent, client):
     check_resp_success(resp)
 
 
-# Disabled until issue #434 can be resolved
-# https://github.com/simonsobs/socs/issues/434
-# @pytest.mark.integtest
-# @inlineCallbacks
-# def test_ibootbar_set_outlet(wait_for_crossbar, start_responder, run_agent, client):
-#     outlet_number = 3
-#     resp = client.set_outlet(outlet=outlet_number, state="on")
-#     check_resp_success(resp)
-#
-#     # Simulate internal state transition of hardware
-#     snmp = SNMPTwister(address, port)
-#     outlet = [("IBOOTPDU-MIB", "outletStatus", outlet_number - 1)]
-#     yield snmp.set(oid_list=outlet, version=2, setvalue=1, community_name="public")
-#
-#     resp = client.acq.start(test_mode=True)
-#     resp = client.acq.wait()
-#
-#     assert resp.session["data"][f"outletStatus_{outlet_number - 1}"]["status"] == 1
+@pytest.mark.integtest
+@inlineCallbacks
+def test_ibootbar_set_outlet(wait_for_crossbar, start_responder, run_agent, client):
+    outlet_number = 3
+    resp = client.set_outlet(outlet=outlet_number, state="on")
+    check_resp_success(resp)
+
+    # Simulate internal state transition of hardware
+    snmp = SNMPTwister(address, port)
+    outlet = [("IBOOTPDU-MIB", "outletStatus", outlet_number - 1)]
+    yield snmp.set(oid_list=outlet, version=2, setvalue=1, community_name="public")
+
+    resp = client.acq.start(test_mode=True)
+    resp = client.acq.wait()
+
+    assert resp.session["data"][f"outletStatus_{outlet_number - 1}"]["status"] == 1
 
 
 @pytest.mark.integtest

--- a/tests/integration/test_ibootbar_agent_integration.py
+++ b/tests/integration/test_ibootbar_agent_integration.py
@@ -10,7 +10,6 @@ from integration.util import create_crossbar_fixture
 from ocs.base import OpCode
 from ocs.testing import create_agent_runner_fixture, create_client_fixture
 from snmpsim.commands import responder
-
 from twisted.internet.defer import inlineCallbacks
 
 from socs.snmp import SNMPTwister

